### PR TITLE
feat: initial track 2 boilerplate

### PR DESCRIPTION
**Pull request by YoDaMa**
This is just the main code that is required from the track 2 template for javascript.

Also included is a simple example demonstrating the resolution API. 

There is no implementation in this PR.

### CODE REVIEW

File: clients/node/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Release History
+## 1.0.9-beta.13 (2020-10-13)
```
Comment (digimaun) [inline clients/node/CHANGELOG.md:2]: It doesn't look like any contents of this file is relevant.

File: clients/node/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Azure IoT Resolver library for Python
```
Comment (cartertinney) [inline clients/node/README.md:1]: Isn't in the Models Repository library (as mentioned in the package.json) or am I mistaken about how the package is organized?

Comment (digimaun) [reply clients/node/README.md:1]: This should probably say for Node

Comment (digimaun) [reply clients/node/README.md:1]: Python stuff goes under `/clients/python`

Comment (YoDaMa) [reply clients/node/README.md:1]: yeah idk why I wrote python. Changing.

File: clients/node/package.json [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  "sdk-type": "client",
+  "main": "dist/index.js",
+  "module": "dist-esm/src/index.js",
+  "browser": {
+    "stream": "./node_modules/stream-browserify/index.js",
+    "./dist-esm/src/print.js": "./dist-esm/src/print.browser.js"
+  },
+  "types": "types/template.d.ts",
+  "scripts": {
+    "audit": "node ../../../common/scripts/rush-audit.js && rimraf node_modules package-lock.json && npm i --package-lock-only 2>&1 && npm audit",
```
Comment (digimaun) [inline clients/node/package.json:14]: Are these scripts, at least for the most part, coming from the track 2 mono repo?

File: clients/node/api-extractor.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{
+  "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
```
Comment (digimaun) [inline clients/node/api-extractor.json:2]: What's the purpose of this file?

File: clients/node/samples/javascript/example.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+/**
+ * Demonstrates something
```
Comment (digimaun) [inline clients/node/samples/javascript/example.js:6]: Demonstrates resolving/obtaining a particular model definition from a remote model repository.

**Comment by digimaun** [comment]
@montgomp

File: clients/node/package.json (hunk 3 above)
Comment (YoDaMa) [reply clients/node/package.json:14]: yes exactly

File: clients/node/api-extractor.json (hunk 4 above)
Comment (digimaun) [reply clients/node/api-extractor.json:2]: Talked offline.

Comment (YoDaMa) [reply clients/node/api-extractor.json:2]: this is specific for track 2 extracting the API to view in microsoft docs.

**Review by digimaun — APPROVED** [review]

**Pull request merged**